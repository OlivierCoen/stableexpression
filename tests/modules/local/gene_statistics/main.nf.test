nextflow_process {

    name "Test Process GENE_STATISTICS"
    script "modules/local/gene_statistics/main.nf"
    process "GENE_STATISTICS"
    tag "genestats"
    tag "module"

    test("Without variation data") {

        when {

            process {
                """
                ch_counts = Channel.fromPath( '$projectDir/tests/data/merge_counts/output/all_counts.parquet', checkIfExists: true).collect()
                ch_metadata = Channel.fromPath( '$projectDir/tests/data/gene_statistics/metadata*.csv', checkIfExists: true ).collect()
                ch_mapping = Channel.fromPath( '$projectDir/tests/data/gene_statistics/mapping*.csv', checkIfExists: true ).collect()
                ch_m_measures = Channel.fromList( [] ).collect()
                input[0] = ch_counts
                input[1] = ch_metadata
                input[2] = ch_mapping
                input[3] = ch_m_measures
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("With m-measures") {

        when {

            process {
                """
                ch_counts = Channel.fromPath( '$projectDir/tests/data/merge_counts/output/all_counts.parquet', checkIfExists: true).collect()
                ch_metadata = Channel.fromPath( '$projectDir/tests/data/gene_statistics/metadata*.csv', checkIfExists: true ).collect()
                ch_mapping = Channel.fromPath( '$projectDir/tests/data/gene_statistics/mapping*.csv', checkIfExists: true ).collect()
                ch_m_measures = Channel.fromPath( '$projectDir/tests/data/pairwise_gene_variation/output/m_measures.csv', checkIfExists: true ).collect()
                input[0] = ch_counts
                input[1] = ch_metadata
                input[2] = ch_mapping
                input[3] = ch_m_measures
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
