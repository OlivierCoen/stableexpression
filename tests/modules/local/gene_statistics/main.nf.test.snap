{
    "Small dataset": {
        "content": [
            {
                "0": [
                    "stats_all_genes.csv:md5,3dd505bec3381b67da2dd316b49e2727"
                ],
                "1": [
                    "stats_most_stable_genes.csv:md5,c07e07ae7d28ccfb63a2fc88f05f77a1"
                ],
                "2": [
                    "log_counts.csv:md5,676b63d29a0658c9c1766021d3d8f862"
                ],
                "3": [
                    [
                        "GENE_STATISTICS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "4": [
                    [
                        "GENE_STATISTICS",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "log_count_summary": [
                    "log_counts.csv:md5,676b63d29a0658c9c1766021d3d8f862"
                ],
                "stats_all_genes": [
                    "stats_all_genes.csv:md5,3dd505bec3381b67da2dd316b49e2727"
                ],
                "stats_most_stable_genes": [
                    "stats_most_stable_genes.csv:md5,c07e07ae7d28ccfb63a2fc88f05f77a1"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-15T09:24:07.245081166"
    },
    "With m-measures": {
        "content": [
            {
                "0": [
                    "stats_all_genes.csv:md5,73aeed3da1d4d396788bdd5ab0c46bd0"
                ],
                "1": [
                    "stats_most_stable_genes.csv:md5,3a770a9b3ccbd5353e7c298d5634790a"
                ],
                "2": [
                    "log_counts.csv:md5,676b63d29a0658c9c1766021d3d8f862"
                ],
                "3": [
                    [
                        "GENE_STATISTICS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "4": [
                    [
                        "GENE_STATISTICS",
                        "polars",
                        "1.17.1"
                    ]
                ],
                "log_count_summary": [
                    "log_counts.csv:md5,676b63d29a0658c9c1766021d3d8f862"
                ],
                "stats_all_genes": [
                    "stats_all_genes.csv:md5,73aeed3da1d4d396788bdd5ab0c46bd0"
                ],
                "stats_most_stable_genes": [
                    "stats_most_stable_genes.csv:md5,3a770a9b3ccbd5353e7c298d5634790a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-15T10:16:21.09953523"
    },
    "Without variation data": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "log_count_summary": [
                    
                ],
                "stats_all_genes": [
                    
                ],
                "stats_most_stable_genes": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-15T10:09:05.24895393"
    }
}